python/bolt-mcp/src/bolt_mcp/oauth/config.py

```
@@ -0,0 +1,153 @@
+
+    async def authenticate(self, username: str, password: str) -> dict | None:
+        """Verify credentials at ``/authorize``. Returns a session dict or ``None``.
+
+        Default: Django's ``authenticate()`` + a new session. Override to use a different
+        credential source or MFA.
+        """
+        from . import sessions  # noqa: PLC0415 — keep Django imports lazy / OAuth optional
+
+        return await sessions.login(username, password)
+
+    async def resolve_user(self, request: Any) -> dict | None:
```

Ruff (0.15.15)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `request`

(ANN401)